src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -166,13 +166,15 @@ long getSizeInBytes() {
             if (protocolBytes != null) {
                 sizeInBytes += protocolBytes.length;
             }
-            if (hdrLen > 0) {
-                sizeInBytes += hdrLen + 2; // CRLF
-            }
-            if (data.length == 0) {
+            sizeInBytes += 2; // CRLF
+            if (!isProtocol()) {
+                if (hdrLen > 0) {
+                    sizeInBytes += hdrLen;
+                }
+                if (dataLen > 0) {
+                    sizeInBytes += dataLen;
+                }

```

Comment:
No, do not touch this. Although there may be improvements available to this code, it should not affect this issue.